src/extensions/dnatco/confal-pyramids/util.ts

```diff
@@ -123,18 +123,23 @@ export namespace ConfalPyramidsUtil {
     export class UnitWalker extends Utility {
         private getAtomIndices(names: string[], residue: Residue): ElementIndex[] {
             let rI = residue.start;
-            const rILast = residue.end - 1;
+            const rILast = residue.end;
             const indices: ElementIndex[] = [];
 
-            for (; rI !== rILast; rI++) {
+            for (; rI <= rILast; rI++) {
```

Comment:
I think this should be just ``< rILast`` since the ranges are always ``[start, end)`` in Mol* (Unless you made it ``[start, end]`` in your implementation).

Reply:
Oops, sorry, looks like I didn't fix the stop condition correctly after changing L126.